### PR TITLE
Initial changes for migrating to `rules_js`

### PR DESCRIPTION
**Pull request by devversion**
See individual commits.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -124,6 +126,14 @@ jobs:
   e2e-windows-subset:
     runs-on: windows-latest
     steps:
+      # Workaround for: https://github.com/bazel-contrib/bazel-lib/issues/968.
```
Comment (josephperrott) [inline .github/workflows/pr.yml:129]: Is this something we would like to have "permanently" in this config?

**Review by josephperrott — APPROVED** [review]
LGTM

File: .github/workflows/pr.yml (hunk 1 above)
Comment (devversion) [reply .github/workflows/pr.yml:129]: There should be a TODO to remove if you're hinting at this 😄

**Review by clydin — APPROVED** [review]

File: packages/angular_devkit/build_angular/src/builders/browser/tests/options/named-chunks_spec.ts [hunk 2]
```diff
@@ -61,6 +61,9 @@ describeBuilder(buildWebpackBrowser, BROWSER_BUILDER_INFO, (harness) => {
       const { result } = await harness.executeOnce();
 
       expect(result?.success).toBe(true);
+      console.error(result);
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/builders/browser/tests/options/named-chunks_spec.ts:64]: Looks like this is still not done.

File: tsconfig.json [hunk 3]
```diff
@@ -26,6 +25,7 @@
       "@angular-devkit/schematics/tools": ["./packages/angular_devkit/schematics/tools/index"],
       "@angular-devkit/schematics/testing": ["./packages/angular_devkit/schematics/testing/index"],
       "@angular-devkit/architect/testing": ["./packages/angular_devkit/architect/testing/index"],
+      "@angular-devkit/build-webpack": ["./packages/angular_devkit/build_webpack"],
```
Comment (alan-agius4) [inline tsconfig.json:28]: Why is this needed? This should be covered by the following line.

File: package.json [hunk 4]
```diff
@@ -176,6 +167,7 @@
     "ora": "5.4.1",
     "pacote": "20.0.0",
     "parse5-html-rewriting-stream": "7.0.0",
+    "patch-package": "^8.0.0",
```
Comment (alan-agius4) [inline package.json:170]: Can't we use `pnpm patch` instead of `patch-package`?

File: yarn.lock [hunk 5]
```diff
@@ -15,31 +15,6 @@ __metadata:
   languageName: node
```
Comment (alan-agius4) [inline yarn.lock:15]: Can you also delete `.yarn` directory, `.yarnrc.yml` and `.yarn` references in the gitignore files?

**Comment by alan-agius4** [comment]
The following commit https://github.com/angular/angular-cli/pull/28957/commits/770405458bde8e8c50562e388dd40fc997e13c7f has an incorrect commit message.

File: package.json (hunk 4 above)
Comment (devversion) [reply package.json:170]: We are in a hybrid state of yarn and pnpm, so we can't unfortunately

File: yarn.lock (hunk 5 above)
Comment (devversion) [reply yarn.lock:15]: We are still using Yarn. see comment above.

File: tsconfig.json (hunk 3 above)
Comment (devversion) [reply tsconfig.json:28]: It didn't work because of the `_`. The directory is `build_webpack`, but the import is `build-webpack`.

File: packages/angular_devkit/build_angular/src/builders/browser/tests/options/named-chunks_spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (devversion) [reply packages/angular_devkit/build_angular/src/builders/browser/tests/options/named-chunks_spec.ts:64]: Note how there was a `debugger` statement left from before 😄

File: package.json [hunk 6]
```diff
@@ -226,8 +218,10 @@
       "built": true
     }
   },
+  "pnpm": {
+    "onlyBuiltDependencies": []
+  },
   "resolutions": {
-    "@bazel/concatjs@npm:5.8.1": "patch:@bazel/concatjs@npm%3A5.8.1#~/.yarn/patches/@bazel-concatjs-npm-5.8.1-1bf81df846.patch",
-    "@microsoft/api-extractor/typescript": "5.6.3"
+    "typescript": "5.6.3"
```
Comment (alan-agius4) [inline package.json:225]: Question, is this still needed?

**Comment by alan-agius4** [comment]
@devversion, I changed the target to minor, as does not apply cleanly on the patch branch.

**Comment by devversion** [comment]
@alan-agius4 Thanks! Our plan is to also submit all these incremental changes to patch ,so I will put together a patch PR

File: package.json (hunk 6 above)
Comment (devversion) [reply package.json]: Yeah, we still need it. Updated to the latest version

**Review by alan-agius4 — APPROVED** [review]

File: WORKSPACE [hunk 7]
```diff
@@ -158,3 +178,49 @@ load("@build_bazel_rules_nodejs//toolchains/esbuild:esbuild_repositories.bzl", "
...
+    strip_prefix = "rules_ts-3.3.1",
+    url = "https://github.com/aspect-build/rules_ts/releases/download/v3.3.1/rules_ts-v3.3.1.tar.gz",
+)
+
+load("@aspect_rules_ts//ts:repositories.bzl", "rules_ts_dependencies")
+
+rules_ts_dependencies(
+    # ts_version_from = "//:package.json",
+    # TODO: Support in https://github.com/aspect-build/rules_ts/blob/main/ts/private/npm_repositories.bzl
+    ts_version = "5.6.2",
```
Comment (clydin) [inline WORKSPACE:219]: Should this be 5.7.2 now?

**Review by clydin — APPROVED** [review]
